beluga/include/beluga/sensor/data/regular_grid.hpp

```diff
@@ -67,15 +67,25 @@ class BaseRegularGrid2 : public ciabatta::ciabatta_top<Derived> {
  public:
   /// Compute nearest grid cell coordinates given plane coordinates.
   /**
-   * Note this is a surjective function.
+   * For performance reasons this function is optimized for coordiantes in the
+   * positive quadrant (i.e. valid grid coordinates). For coordinates in other
+   * quadrants a representative cell in the same quadrant will be returned instead.
    *
    * \param x Plane x-axis coordinate.
    * \param y Plane y-axis coordinate.
    * \return Grid cell coordinates.
    */
   [[nodiscard]] Eigen::Vector2i cell_near(double x, double y) const {
-    const auto xi = static_cast<int>(std::floor(x / this->self().resolution()));
-    const auto yi = static_cast<int>(std::floor(y / this->self().resolution()));
+    const auto inv_resolution = 1 / this->self().resolution();
```

Comment:
@glpuga nit: avoid implicit promotion
```suggestion
    const auto inv_resolution = 1.0 / this->self().resolution();
```

Reply:
Fixed.